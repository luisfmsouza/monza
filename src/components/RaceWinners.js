import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const List = styled.ul`
  padding: 0;
  margin: 0;
`;

const Item = styled.li`
  display: grid;
  grid-template-columns: 80px 30px auto;
  grid-template-areas:
    "date dot name"
    "date dot circuit";
  margin: 30px 0;
`;

const RaceDate = styled.span`
  align-self: center;
  font-size: 12px;
  grid-area: date;
  text-align: right;
`;

const Dot = styled.span`
  align-self: center;
  background-color: white;
  border-radius: 15px;
  grid-area: dot;
  justify-self: center;
  height: 7px;
  width: 7px;
`;

const Name = styled.span`
  font-size: 24px;
  font-weight: bold;
  grid-area: name;
`;

const Circuit = styled.span`
  font-size: 12px;
  grid-area: circuit;
`;

const RaceWinners = ({ raceListBySeason, season, champion }) =>
  raceListBySeason && (
    <List>
      {raceListBySeason.races.map((race, index) => (
        <Item key={`${season}-${index}`}>
          <RaceDate>{race.date}</RaceDate>
          <Dot />
          <Name>
            {`${race.givenName} ${race.familyName}`}
            {race.code === champion && " üèÜ"}
          </Name>
          <Circuit>{race.circuitName}</Circuit>
        </Item>
      ))}
    </List>
  );

RaceWinners.propTypes = {
  raceListBySeason: PropTypes.shape({
    season: PropTypes.string.isRequired,
    races: PropTypes.arrayOf(
      PropTypes.shape({
        driverId: PropTypes.string.isRequired,
        code: PropTypes.string.isRequired,
        url: PropTypes.string.isRequired,
        givenName: PropTypes.string.isRequired,
        familyName: PropTypes.string.isRequired,
        dateOfBirth: PropTypes.string.isRequired,
        nationality: PropTypes.string.isRequired,
        circuitName: PropTypes.string,
        date: PropTypes.string
      })
    ).isRequired
  }),
  season: PropTypes.string.isRequired,
  champion: PropTypes.string.isRequired
};

export default RaceWinners;
