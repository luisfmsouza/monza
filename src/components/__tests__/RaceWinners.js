import React from "react";
import { shallow } from "enzyme";

import RaceWinners from "../RaceWinners";

describe("RaceWinners", () => {
  it("renders without crashing", () => {
    const props = {
      raceListBySeason: {
        season: "2005",
        races: [
          {
            driverId: "fisichella",
            code: "FIS",
            url: "http://en.wikipedia.org/wiki/Giancarlo_Fisichella",
            givenName: "Giancarlo",
            familyName: "Fisichella",
            dateOfBirth: "1973-01-14",
            nationality: "Italian",
            circuitName: "Albert Park Grand Prix Circuit",
            date: "2005-03-06"
          },
          {
            driverId: "alonso",
            permanentNumber: "14",
            code: "ALO",
            url: "http://en.wikipedia.org/wiki/Fernando_Alonso",
            givenName: "Fernando",
            familyName: "Alonso",
            dateOfBirth: "1981-07-29",
            nationality: "Spanish",
            circuitName: "Bahrain International Circuit",
            date: "2006-03-12"
          }
        ]
      },
      season: "2005",
      champion: "ALO"
    };
    const wrapper = shallow(<RaceWinners {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
