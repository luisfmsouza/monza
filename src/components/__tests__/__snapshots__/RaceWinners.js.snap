// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RaceWinners renders without crashing 1`] = `
<styled.ul>
  <styled.li
    key="2005-0"
  >
    <styled.span>
      2005-03-06
    </styled.span>
    <styled.span />
    <styled.span>
      Giancarlo Fisichella
    </styled.span>
    <styled.span>
      Albert Park Grand Prix Circuit
    </styled.span>
  </styled.li>
  <styled.li
    key="2005-1"
  >
    <styled.span>
      2006-03-12
    </styled.span>
    <styled.span />
    <styled.span>
      Fernando Alonso
       üèÜ
    </styled.span>
    <styled.span>
      Bahrain International Circuit
    </styled.span>
  </styled.li>
</styled.ul>
`;
